homeassistant/components/zone/__init__.py

```diff
@@ -141,6 +149,28 @@ def async_active_zone(
     return closest
 
 
+@callback
+def async_setup_track_zone_entity_ids(hass: HomeAssistant) -> None:
+    """Set up track of entity IDs for zones."""
+    zone_entity_ids: list[str] = hass.states.async_entity_ids(DOMAIN)
+    hass.data[ZONE_ENTITY_IDS] = zone_entity_ids
+
+    @callback
+    def _async_add_zone_entity_id(event_: Event) -> None:
+        """Add zone entity ID."""
+        zone_entity_ids.append(event_.data[ATTR_ENTITY_ID])
+        zone_entity_ids.sort()
```

Comment:
These have to be sorted for deterministic order but we only have to do it when one is added or removed which is far cheaper

https://github.com/home-assistant/core/pull/94866/files#diff-69e20858d88c5e35bee0e5e8671479e8040bcb8488cea7deee3fecef4b6a850eL115